### ISSUE TITLE
Parse.Promise.when callback arguments should match the format of promises passed in

### ISSUE THREAD

**Issue by braker1nine** [ParsePlatform/Parse-SDK-JS#114]
For example
```javascript
var p1 = Parse.Promise.as('Thing1');
var p2 = Parse.Promise.as('Thing2');
var promiseArray = [p1,p2];
Parse.Promise.when(promiseArray).then(function(results) {
  console.log(results); // should be an array containing 'Thing1' and 'Thing2'
});
```

**Comment by andrewimm** [comment]
Are you suggesting that when passing promises as arguments, the result should return them as arguments, and when calling as an array it should return them as an array?

If so, I agree. That should be a pretty easy fix.

**Comment by braker1nine** [comment]
@andrewimm yep that's exactly what I'm suggesting. Would make it much simpler to deal with a dynamic number of promises. 

Unless you'd prefer to, I can put together a PR. Just didn't want to do it if this was an intentional design decision.

**Comment by andrewimm** [comment]
If you want to go ahead with a PR, please do. We're a bit busy with the new site launch today :)

**Comment by TylerBrock** [comment]
Hey, no big deal here, just a BACKWARD BREAKING change that is not documented anywhere... no change log or anything.

**Comment by andrewimm** [comment]
@TylerBrock you're absolutely right. I should have given this more consideration. How do you suggest we handle this?

**Comment by andrewimm** [comment]
I'm considering paths forward. In examining the developer benefits of this change, I neglected to realize that it would break past apps.

As great as it would be to simplify this, my current plan is to revert this commit and remove 1.6.10 from npm.

**Comment by andrewimm** [comment]
Reverted and published as 1.6.11.

That said, Tyler, if these sorts of changes are affecting you in production, you should really lock down your package.json to reference explicit version numbers.

**Comment by braker1nine** [comment]
Understand not wanting to make breaking changes. Sorry for the inconvenience @TylerBrock

I still think it's a clear improvement that should be used. Any policy for that? Documentation-wise I screwed up. Changes from #116 should have been included in the PR

**Comment by TylerBrock** [comment]
This actually solves a pain point for us and we love this change. No issues there. It actually didn't affect us as we are explicit with package versions. I was just concerned that the change was not communicated at all. Unless you watch this repo like a hawk you'd have no idea why code that works against 1.6.9 is totally broken in 1.6.10.

**Comment by lacker** [comment]
I agree with Tyler, let's consider this an error and try to avoid this level of breakery in the future.

**Comment by andrewimm** [comment]
While the project is not semantically vendored, this is definitely a change that should go in a minor version bump, not a patch. That seems the safest route.

**Comment by TylerBrock** [comment]
Great, thanks all!

**Comment by braker1nine** [comment]
@andrewimm @lacker can we reopen this issue since it seems to be generally agreed that it is an issue in need of improvement, even if the deployment wasn't handled optimally?

**Comment by fatuhoku** [comment]
I concur with @braker1nine. Took me a while to navigate to this issue — I even put it up on Stackoverflow.

http://stackoverflow.com/questions/34813008/how-can-i-fetch-specific-rows-from-a-parse-class-with-indexes-parse-promise-wh

As I understand it, the behaviour as described by @braker1nine above is only available in 1.7? or what? What are the affected versions, where `Parse.Promise.when` doesn't return array results for array arguments?
